### PR TITLE
Complete view all, search results page frontend. Add search bar to all navbars and store user input.

### PR DESCRIPTION
**Pull request by smruthibalajee**
**Description:** Completed the frontend for the business gallery page that will dynamically load content from both the view all feature of carousels and the search bar. Added two different types of search bars to the nav bars, depending on the type of page. Organized html and javascript files by naming them appropriately (in a way that describes the behavior of the functions) and splitting all the relevant code for each html file into a different script file.

**Key features:**
- Search bar: When the user presses enter, stores the user input string in localStorage (currently prints to console). This will then be used to dynamically load the name of the business gallery page (the string that the user searches). After special characters are removed, it will also be used to query search results for the gallery page.

- Business gallery page: Utilized bootstrap to create a dynamically resizing card gallery. When the screen size is reduced, the number of rows and columns changes. Additionally, when new HTML card elements are created, it will automatically wrap the card to the next row if the current one exceeds 4 cards. This will be very useful when designing the dynamic loading from database feature for the page. Currently all cards are hardcoded, but will add the backend firebase query next.

View all page demo:
viewall

### CODE REVIEW

File: ripple/src/main/webapp/businesscarousels.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/* Dynamic Carousel functions. */
+
+/* Function that takes the user inputted address and converts it to latitude and longitude coordinates. */
+function convertAddressToCoord() {
+  var address = localStorage.getItem('inputAddress');
+  var geocoder = new google.maps.Geocoder();
+  console.log("old" + localStorage.getItem('enteredLat') + localStorage.getItem('enteredLong'));
+  geocoder.geocode( { 'address': address}, function(results, status) {
```
Comment (peter14f) [inline ripple/src/main/webapp/businesscarousels.js:8]:
```
geocoder.geocode( { 'address': address}, (results, status) => {
```

File: ripple/src/main/webapp/businesscarousels.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+//TODO: use latitude and longitude to query businesses within a certain radius in future, when Firestore allows for Geopoint queries
+/* Given a latitude and longitude, finds the name of the city in the US with those coordinates. */
+function convertCoordToCity() {
+  var geocoder = new google.maps.Geocoder();
+  console.log(localStorage.getItem('enteredLat')+ ' in coordToCity');
+  console.log(localStorage.getItem('enteredLong') + ' in coordToCity');
+  var latlng = { lat: parseFloat(localStorage.getItem('enteredLat')), lng: parseFloat(localStorage.getItem('enteredLong'))};
+  geocoder.geocode({ location: latlng }, function(results, status) {
```
Comment (peter14f) [inline ripple/src/main/webapp/businesscarousels.js:29]:
```
geocoder.geocode({ location: latlng }, (results, status) => {
```

File: ripple/src/main/webapp/landing.js [hunk 3]
```diff
@@ -33,9 +33,9 @@ function searchAddressAutocomplete() {
     });
 }
 
+var ENTER_KEY = 13;
```
Comment (peter14f) [inline ripple/src/main/webapp/landing.js:36]: There's 2 ENTER_KEY global variables now. Could we consolidate to just having one?

File: ripple/src/main/webapp/script.js [hunk 4]
```diff
@@ -74,6 +74,24 @@ function selectFile(fileId) {
...
+    if (keyPress.keyCode === ENTER_KEY) {
+      var searchString = document.getElementById('search-bar').value;
+      console.log(searchString);
+      localStorage.setItem('searchString', searchString);
+      localStorage.setItem('galleryPageName', searchString);
+      window.location.assign("businessgallery.html");
+    } else {
+        return false;
+    }
+
```
Comment (peter14f) [inline ripple/src/main/webapp/script.js:94]: Is the closing bracket missing for searchInput()? Also indentation seems off for this method.

File: ripple/src/main/webapp/businesscarousels.js (hunk 2 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/businesscarousels.js]: fixed!

File: ripple/src/main/webapp/businesscarousels.js (hunk 1 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/businesscarousels.js]: fixed!

File: ripple/src/main/webapp/landing.js (hunk 3 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/landing.js]: Yes, sorry I didn't realize it was in a different file!

File: ripple/src/main/webapp/script.js (hunk 4 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/script.js:95]: I must have messed this up when merging! Just fixed.

File: ripple/src/main/webapp/businesscarousels.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
+/* Dynamic Carousel functions. */
+
+/* Function that takes the user inputted address and converts it to latitude and longitude coordinates. */
+function convertAddressToCoord() {
+  var address = localStorage.getItem('inputAddress');
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/businesscarousels.js:5]: Use double quotation to match syntax from rest of code

File: ripple/src/main/webapp/businesscarousels.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  var geocoder = new google.maps.Geocoder();
+  console.log("old" + localStorage.getItem('enteredLat') + localStorage.getItem('enteredLong'));
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/businesscarousels.js:7]: Use double quotation to match syntax from rest of code

File: ripple/src/main/webapp/businesscarousels.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  geocoder.geocode( { 'address': address}, (results, status) => {
+    if (status == 'OK') {
+      //Stores user's latitude and longitude in localStorage, to be used for computing walking distance
+      localStorage.setItem('enteredLat', results[0].geometry.location.lat());
+      localStorage.setItem('enteredLong', results[0].geometry.location.lng());
+      console.log(localStorage.getItem('enteredLat'));
+      console.log(localStorage.getItem('enteredLong'));
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/businesscarousels.js:14]: dbl quotes

File: ripple/src/main/webapp/businesscarousels.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    }
+  });
+}
+
+//TODO: use latitude and longitude to query businesses within a certain radius in future, when Firestore allows for Geopoint queries
+/* Given a latitude and longitude, finds the name of the city in the US with those coordinates. */
+function convertCoordToCity() {
+  var geocoder = new google.maps.Geocoder();
+  console.log(localStorage.getItem('enteredLat')+ ' in coordToCity');
+  console.log(localStorage.getItem('enteredLong') + ' in coordToCity');
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/businesscarousels.js:27]: dbl quotes

File: ripple/src/main/webapp/businesscarousels.js [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+          carouselInner.firstElementChild.className += " active";
+          $(carouselElement).carousel({slide : true, interval : false});
+        }
+   });
+}
+
+/* When view all is clicked, redirects to the gallery page and sets the name to be view all. */
+function viewAll() {
+  localStorage.setItem('galleryPageName', 'View all');
+  window.location.assign("businessgallery.html");
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/businesscarousels.js:127]: nit dbl quotes. Apply to rest of code

File: ripple/src/main/webapp/managebusinessinfo.html [hunk 10]
```diff
@@ -28,13 +27,91 @@
     </script>
     <!-- This needs to be after firebase initialization -->
     <script src="auth.js"></script>
+    <script src="firestore.js"></script>
```
Comment (saraholiviawu) [inline ripple/src/main/webapp/managebusinessinfo.html:30]: delete

File: ripple/src/main/webapp/businesscarousels.js (hunk 5 above)
Comment (smruthibalajee) [reply ripple/src/main/webapp/businesscarousels.js:5]: changed to all double quotes

**Pull request merged**